### PR TITLE
Upload erratum

### PR DESCRIPTION
**Pull request by Ichimonji10**
The first commit performs a simple refactor, where a private function is moved into a `utils` module and made public. The second commit is where the real action happens: a brand new test case is added!

### CODE REVIEW

**Comment by Ichimonji10** [comment]
Depends on https://github.com/PulpQE/pulp-fixtures/pull/14

**Comment by Ichimonji10** [comment]
I tested this change by going to the Pulp Fixtures repository and executing the following:

```sh
make fixtures/rpm-erratum
python -m http.server
```

I then applied this diff to Pulp Smash:

```diff
diff --git a/pulp_smash/constants.py b/pulp_smash/constants.py
index 5e0d2a5..43035f4 100644
--- a/pulp_smash/constants.py
+++ b/pulp_smash/constants.py
@@ -217,8 +217,9 @@ RPM_ABS_PATH = (
 """The absolute path to :data:`pulp_smash.constants.RPM` in the filesystem."""
 
 RPM_ERRATUM_URL = (
-    'https://repos.fedorapeople.org'
-    '/repos/pulp/pulp/fixtures/rpm-erratum/erratum.json'
+    'http://localhost:8000/fixtures/rpm-erratum/erratum.json'
+    # 'https://repos.fedorapeople.org'
+    # '/repos/pulp/pulp/fixtures/rpm-erratum/erratum.json'
 )
 """The URL to an JSON erratum file for an RPM repository."""
 
```

Finally, I executed `python -m unittest2 pulp_smash.tests.rpm.api_v2.test_upload_publish.UploadErratumTestCase`, for both Pulp 2.8.5 and master (2.9).

**Comment by Ichimonji10** [comment]
Closes #253.

File: pulp_smash/tests/rpm/api_v2/test_upload_publish.py [hunk 1]
```diff
@@ -13,19 +13,138 @@
...
 
 
+def parse_updateinfo_update(update):
+    """Create an erratum from an ``updateinfo.xml`` update.
+
+    This function creates an empty erratum, then parses a top-level ``update``
+    section from an ``updateinfo.xml`` file, and populates the empty erratum
+    from such.
+
+    .. WARNING:: There is no schema for ``updateinfo.xml`` files. Instead, this
```
Comment (elyezer) [inline pulp_smash/tests/rpm/api_v2/test_upload_publish.py]: > There is no schema for ``updateinfo.xml`` files.

:disappointed:

File: pulp_smash/tests/rpm/api_v2/test_upload_publish.py [hunk 2]
```diff
@@ -13,19 +13,138 @@
...
+        code. As a result, this function may exhibit erroneous behaviour.
+
+        As of this writing, the DNF source code can be cloned from
+        ``https://github.com/rpm-software-management/dnf.git``. The files used
+        by this function are:
+
+        * `dnf/cli/commands/updateinfo.py`_
+        * `tests/repos/rpm/updateinfo.xml`_
+
+    .. NOTE:: DNF is incompatible with the openSUSE ``updateinfo.xml`` schema.
```
Comment (elyezer) [inline pulp_smash/tests/rpm/api_v2/test_upload_publish.py]: Great investigation. At least seems that OpenSUSE is trying to make its way to a standard of specification for the file.

File: pulp_smash/tests/rpm/api_v2/test_upload_publish.py [hunk 3]
```diff
@@ -13,19 +13,138 @@
...
+        "from," and "id."
+
+    .. _dnf/cli/commands/updateinfo.py:
+        https://github.com/rpm-software-management/dnf/blob/master/dnf/cli/commands/updateinfo.py
+    .. _tests/repos/rpm/updateinfo.xml:
+        https://github.com/rpm-software-management/dnf/blob/master/tests/repos/rpm/updateinfo.xml
+    .. _openSUSE:Standards Rpm Metadata UpdateInfo:
+        https://en.opensuse.org/openSUSE:Standards_Rpm_Metadata_UpdateInfo
+    """
+    erratum = {}
```
Comment (elyezer) [inline pulp_smash/tests/rpm/api_v2/test_upload_publish.py]: The below should make easy to add more items and avoid repeating the `erratum` variable. This is not a blocker, feel free to update it or not.

```python
erratum = {
    'pkglist': _get_pkglist(update),
    'reboot_suggested': _get_reboot_suggested(update)
    'references': _get_references(update),
}

File: pulp_smash/tests/rpm/api_v2/test_upload_publish.py (hunk 2 above)
Comment (Ichimonji10) [inline pulp_smash/tests/rpm/api_v2/test_upload_publish.py]: Thanks!

Research was a huge part of this PR. It took a while to find out where errata come from and what they should look like, and where `updateinfo.xml` files go and what they should look like.

**Pull request merged**